### PR TITLE
DOC: Fix flake8 issues in doc/source/whatsnew/v0.17.*.rst

### LINKED ISSUE #24235: DOC: Fix flake8 issues in doc/source/whatsnew/v0.17.*.rst
xref: #24177 

We didn't start validating the format of PEP8 and other code standards in the documentation examples until recently. We still have some files with errors, that we need to skip, and that we should fix, so we can start validating them.

**Two steps are required to replicate the issue**

1. Edit `setup.cfg` in the pandas home, and in the flake8-rst section, remove from the `exclude` list the file `doc/source/whatsnew/v0.17.0.rst` & `doc/source/whatsnew/v0.17.1.rst`
2. Remove the import block ( `from pandas import *`) from the lines (8-11) of the file `v0.17.*.rst`. 

After that, running the next command will report the errors in the file (note that syntax error usually prevent to validate other errors, and the list of errors to fix can become much longer when the syntax error is fixed (please make sure that you are using `flake8-rst` version 0.7.0 or higher):

```
$ flake8-rst doc/source/whatsnew/v0.17.*
doc/source/whatsnew/v0.17.0.rst:83:23: E203 whitespace before ':'
doc/source/whatsnew/v0.17.0.rst:83:47: E231 missing whitespace after ','
doc/source/whatsnew/v0.17.0.rst:84:23: E203 whitespace before ':'
doc/source/whatsnew/v0.17.0.rst:84:47: E231 missing whitespace after ','
doc/source/whatsnew/v0.17.0.rst:84:57: E231 missing whitespace after ','
doc/source/whatsnew/v0.17.0.rst:85:23: E203 whitespace before ':'
doc/source/whatsnew/v0.17.0.rst:85:47: E231 missing whitespace after ','
doc/source/whatsnew/v0.17.0.rst:85:57: E231 missing whitespace after ','
doc/source/whatsnew/v0.17.0.rst:110:39: E231 missing whitespace after ','
doc/source/whatsnew/v0.17.0.rst:110:49: E231 missing whitespace after ','
doc/source/whatsnew/v0.17.0.rst:115:39: E231 missing whitespace after ','
doc/source/whatsnew/v0.17.0.rst:115:49: E231 missing whitespace after ','
doc/source/whatsnew/v0.17.0.rst:122:31: E231 missing whitespace after ','
doc/source/whatsnew/v0.17.0.rst:122:41: E231 missing whitespace after ','
doc/source/whatsnew/v0.17.0.rst:123:31: E231 missing whitespace after ','
doc/source/whatsnew/v0.17.0.rst:123:41: E231 missing whitespace after ','
doc/source/whatsnew/v0.17.0.rst:141:25: E203 whitespace before ':'
doc/source/whatsnew/v0.17.0.rst:141:47: E231 missing whitespace after ','
doc/source/whatsnew/v0.17.0.rst:141:55: E231 missing whitespace after ','
...
doc/source/whatsnew/v0.17.1.rst:82:11: F821 undefined name 'DataFrame'
doc/source/whatsnew/v0.17.1.rst:82:25: E203 whitespace before ':'
doc/source/whatsnew/v0.17.1.rst:82:35: E226 missing whitespace around arithmetic operator
```

Once all the errors are addressed, please open a pull request with the fixes in the file, and removing the file from setup.cfg. If you need to do something that feels wrong to fix an error, please ask in a comment to this issue. Please avoid other unrelated changes, which can be addressed in a separate pull request.
cc: @datapythonista

### PR DESCRIPTION
**Pull request by aqurilla**
- [x] closes #24235
- [ ] tests added / passed
- [ ] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

### CODE REVIEW

**Comment by aqurilla** [comment]
Hi! @datapythonista 
I have a few queries regarding some changes - currently the output of 
`flake8-rst doc/source/whatsnew/v0.17.*` is:

doc/source/whatsnew/v0.17.0.rst:177:22: E225 missing whitespace around operator
doc/source/whatsnew/v0.17.0.rst:177:22: E999 SyntaxError: invalid syntax
doc/source/whatsnew/v0.17.0.rst:177:26: E225 missing whitespace around operator
doc/source/whatsnew/v0.17.0.rst:750:14: E711 comparison to None should be 'if cond is None:'
doc/source/whatsnew/v0.17.0.rst:757:6: E711 comparison to None should be 'if cond is None:'
doc/source/whatsnew/v0.17.0.rst:772:12: E711 comparison to None should be 'if cond is None:'
doc/source/whatsnew/v0.17.1.rst:87:11: F821 undefined name 'DataFrame'

The errors on line 177 are due to `df.plot.<TAB>` - which is a valid operation in pandas but is giving an error here.

The next 3 errors are due to a comparison of a series vs None: `s == None`,
but an error is thrown requesting an `if` loop be added, along with its indentation block.

What would be the best way to resolve these?

**Comment by datapythonista** [comment]
we fixed them before with a `noqa` comment. If you do a `grep "<TAB>" doc/*`, `grep "== None"`... you should find examples.

**Comment by datapythonista** [comment]
also, please merge master and fix the conflicts when you update the PR

**Comment by datapythonista** [comment]
@aqurilla do you have time to finish this PR?

**Comment by aqurilla** [comment]
Hi @datapythonista yes, I was travelling and should be able to finish this by tomorrow

**Comment by datapythonista** [comment]
@aqurilla do you have time to finish this?

**Comment by aqurilla** [comment]
Hi @datapythonista , please have a look,
sorry for the delay!

File: doc/source/whatsnew/v0.17.0.rst [hunk 1]
```diff
@@ -748,14 +747,14 @@ Previous Behavior:
 
 .. code-block:: ipython
 
-   In [5]: s==None
+   In [5]: s == None  # noqa: E711
```
Comment (jreback) [inline doc/source/whatsnew/v0.17.0.rst]: we could ignore this E711 in the docs (but leave for code & tests itself), is this possible to discriminate like this? @datapythonista

Comment (datapythonista) [reply doc/source/whatsnew/v0.17.0.rst]: yes, in `setup.cfg` we have different configurations for `flake8` (code) and `flake8-rst` (docs). SOunds good to me.

@aqurilla can you ignore `E711` for `flake8-rst` there, and remove the `noqa` here please?

Comment (aqurilla) [reply doc/source/whatsnew/v0.17.0.rst]: @datapythonista sure, please have a look,

Comment (jreback) [reply doc/source/whatsnew/v0.17.0.rst]: looks good

**Comment by jreback** [comment]
ping on green.

**Pull request merged**